x-pack/docs/en/security/authorization/run-as-privilege.asciidoc

```diff
@@ -2,34 +2,181 @@
-be <<ref-ad-settings,configured with a `bind_dn` and `secure_bind_password`>> to
-support _run as_. The Kerberos, OpenID Connect, PKI, and SAML realms do not
-support _run as_.
+supports the _run as_ mechanism. The internal `native` and `file` realms
+inherently support this capability, as does the JWT realm. To support `run_as`,
+in the LDAP realm, you have to run in
+<<ldap-realm-configuration,_user search_ mode>>. For Active Directory, you need
+to <<ref-ad-settings,configure a `bind_dn` and `secure_bind_password`>>.
 
 To submit requests on behalf of other users, you need to have the `run_as`
-permission. For example, the following role grants permission to submit request
-on behalf of `jacknich` or `redeniro`:
+privilege in your <<defining-roles,roles>>. For example, the following request
+creates a `my_director` role that grants permission to submit request on behalf
+of `jacknich` or `redeniro`:
 
-[source,js]
----------------------------------------------------
+[source,console]
+----
+POST /_security/role/my_director?refresh=true
 {
-  "run_as" : [ "jacknich", "rdeniro" ]
+  "cluster": ["manage"],
+  "indices": [
+    {
+      "names": [ "index1", "index2" ],
+      "privileges": [ "manage" ]
+    }
+  ],
+  "run_as": [ "jacknich", "rdeniro" ],
+  "metadata" : {
+    "version" : 1
+  }
 }
----------------------------------------------------
+----
 
 To submit a request as another user, you specify the user in the
 `es-security-runas-user` request header. For example:
 
-[source,shell]
----------------------------------------------------
-curl -H "es-security-runas-user: jacknich"  -u es_admin -XGET 'http://localhost:9200/'
----------------------------------------------------
+[source,sh]
+----
+curl -H "es-security-runas-user: jacknich" -u es-admin -X GET http://localhost:9200/
+----
+
+The `run_as` user passed in through the `es-security-runas-user` header must be
+a username from a realm that supports lookup. Because the Kerberos, OpenID
+Connect, PKI, and SAML realms don't support user lookup, these realms can't use
+`run_as` directly. However, users that authenticate from one of these realms can
+run requests on behalf of users from the `native`, `file`, LDAP, and Active
+Directory realms. For example, an OIDC user can authenticate to {es} and submit
+requests as a `run_as` user in the `native` realm. {es} will retrieve the
+indicated `runas` user and submit the request using their roles.
+
+[[run-as-privilege-apply]]
+== Apply the `run_as` privilege to roles
+You can apply the `run_as` privilege when creating roles with the
+<<security-api-put-role,create or update roles API>>. Users who are assigned
+a role that contains the `run_as` privilege inherit all privileges from their
+role, and can also submit requests on behalf of the indicated users.
+
+The `run_as` privilege essentially operates like a secondary form of
+<<authorization_realms,delegated authorization>>. After a user successfully
```

Comment:
A key difference is:
- Delegated authorization applies to the first user who is authenticated.
- Run-as applies to the second user who is being impersonated.

A side goal for me is to figure out a way to call them both Delegated Authorization, but with qualifiers that help illustrate the similarities and differences. Some ideas might be:
- Primary User Delegated Authorization, Authenticated User Authorization Delegation
- Secondary User Delegated Authorization, Run-as Delegated Authorization, Impersonated User Authorization Delegation